File: playwright/pages/components/UserDetailsDrawer.js
import { BasePage } from "../BasePage.js";

export class UserDetailsDrawer extends BasePage {
  async close() {
    await this.testId("details-close").click();
  }

  async isLoaded() {
    return (await this.isVisible("details-name")) && (await this.isVisible("details-email"));
  }
}

File: src/test/java/com/automation/um/pages/components/UserDetailsDrawer.java
package com.automation.um.pages.components;

import com.automation.um.pages.BasePage;
import org.openqa.selenium.WebDriver;

public class UserDetailsDrawer extends BasePage {
  public UserDetailsDrawer(WebDriver driver) {
    super(driver);
  }

  public void close() {
    el("details-close").click();
  }

  public boolean isLoaded() {
    return isVisible("details-name") && isVisible("details-email");
  }
}
